enhancements/update/distribute-secondary-metadata-as-container-image.md

```diff
@@ -0,0 +1,184 @@
+This image will be available for public consumption, and will be used by the publicly available Cincinnati instances operated by Red Hat.
+The image can be transferred and served to Cincinnati instances operated by any party, including but not limited to network-restricted environments.
+The source of the container image, namely the cincinnati-graph-data, will remain publicly available and its workflows unaffected by this enhancement.
+
+## Motivation
+
+### Goals
+1. The container image will become the primary way of consuming and distributing the content of the cincinnati-graph-data repository.
+2. As long as the primary release metadata will be packaged, distributed and consumed in form of container images as well, the transport mechanism required for data consumption by Cincinnati, as well as mirroring and serving its consumed content, will be solely [Docker Registry HTTP API V2][docker-registry-http-api-v2] protocol.
+
+
+### Non-Goals
+1. Changes to the GitHub repository workflow or data format and make progress.
+2. Implementation details about the streamlining of the mirroring process for primary and secondary release metadata.
+
+## Proposal
+
+### User Stories
+
+#### Content mirroring for Cincinnati deployments
+A user might want to mirror all content which is required by Cincinnati.
+
+1. The user sets up a container image registry in the target environment.
+2. Optionally, if the target environment does not have internet access, the user downloads the release container images and the desired graph-data container image to a transfer medium.
+3. The user pushes the release and graph-data images to the target container image registry
+4. The user can now configure the Cincinnati instance in the target environment to consume the content from the available container image registry
+
+### Implementation Details/Notes/Constraints
+
+#### CPaaS Build configuration
+1. Every change to the master branch of the cincinnati-graph-data results in a new container image containing the unmodified working tree files of the new revision.
+  This will allow the plugin to continuously grab the latest version without being informed that a latest version exists.
+
+2. The image will be pushed to a publicly available container image registry under a well-known repository name, and it's `latest` tag will be updated to the latest revision.
+
+#### Plugin implementation
+1. A new Cincinnati plugin will be implemented to fetch the container image from a configured container image repository.
+  The registry, repository name, a tag or a digest will be configurable.
+  If a tag is specified, the plugin will check if the specified tag changed on each plugin execution, and download the content if that check is positive.
+
+2. The Git(Hub) scraper plugin will be deprecated in favor of the new plugin. Note that Cincinnati has support to use no cincinnati-graph-data scraper plugin at all, and can be provided the content to the filesystem directly by the its operator.
```

Comment:
We need a back mechanism to feed the production Cincinnati with secondary metadata if the image building pipeline gets in to some issue. Because the CPaaS pipeline is not under our control .So we need some backup mechanism to keep feeding Cincinnati running in production by APP- SRE. 

If we deprecate the Git or Github scrapper what option we have for an alternate mechanism?

Reply:
The same option as the Cincinnati Operator uses today. Download the data from GitHub, extract it to the filesystem, mount the path to a volume in the Cincinnati Graph-Builder, and configure it to use that path directly and not scrape any secondary metadata source.